test/testapps/TestServer/Startup.cs

```diff
@@ -39,7 +40,14 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
             // Mount the server-side Blazor app on /subdir
             app.Map("/subdir", subdirApp =>
             {
-                subdirApp.UseServerSideBlazor<BasicTestApp.Startup>();
+                // The following two lines are equivalent to:
+                //     subdirApp.UseServerSideBlazor<BasicTestApp.Startup>();
+                // However it's expressed using UseSignalR+UseBlazor as a way of checking that
+                // we're not relying on any extra magic inside UseServerSideBlazor, since it's
+                // important that people can set up these bits of middleware manually (e.g., to
+                // swap in UseAzureSignalR instead of UseSignalR).
+                subdirApp.UseSignalR(route => route.MapHub<BlazorHub>(BlazorHub.DefaultPath));
+                subdirApp.UseBlazor<BasicTestApp.Startup>();
```

Comment:
I love it. I 100% agree with the analysis and I think this is the right solution.